frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx

```diff
@@ -0,0 +1,342 @@
+import React from "react";
+import { AxisBottom, AxisLeft } from "@visx/axis";
+import { GridRows } from "@visx/grid";
+import { scaleBand, scaleLinear } from "@visx/scale";
+import { Bar } from "@visx/shape";
+import { Group } from "@visx/group";
+import { Text } from "@visx/text";
+import { assoc, merge } from "icepick";
+import {
+  getLabelProps,
+  getXTickLabelProps,
+  getXTickWidth,
+  getYTickLabelProps,
+  getYTickWidth,
+} from "metabase/static-viz/lib/axes";
+import { formatNumber } from "metabase/static-viz/lib/numbers";
+import {
+  calculateWaterfallDomain,
+  calculateWaterfallEntries,
+  calculateWaterfallSeriesForValues,
+  formatTimescaleWaterfallTick,
+  getWaterfallEntryColor,
+} from "metabase/static-viz/lib/waterfall";
+import { measureTextHeight, truncateText } from "metabase/static-viz/lib/text";
+import { sortTimeSeries } from "../../lib/sort";
+import {
+  DATE_ACCESSORS,
+  POSITIONAL_ACCESSORS,
+} from "../../constants/accessors";
+import { getWaterfallColors } from "../../lib/colors";
+import Values from "../Values";
+import { createXScale } from "../XYChart/utils";
+
+const layout = {
+  width: 540,
+  height: 300,
+  margin: {
+    // Add some margin so when the chart scale down,
+    // elements that are rendered at the top of the chart doesn't get cut off
+    top: 12,
+    left: 55,
+    right: 40,
+  },
+  barPadding: 0.2,
+  labelPadding: 12,
+  strokeDasharray: "4",
+  numTicks: 4,
+  maxTickWidth: 100,
+};
+
+// If inner chart area is smaller than this then it will look cramped
+const MIN_INNER_HEIGHT = 250;
+const MAX_EXTRA_HEIGHT = 40;
```

Comment:
in `<XYChart />` we can add 40px for showing legend, so I think it's safe to allow expanding chart to the maximum of 40px to allow more room for rendering the chart since now we also rotate the x-axis tick label and always show the x-axis label the x-axis height could be up to around 140px tall.
![image](https://user-images.githubusercontent.com/1937582/191934657-75720c5b-9a42-46d3-8ecf-751fc7d0e4fd.png)

This makes it's height relatively the same

#### After
![Screen Shot 2022-09-23 at 4 49 40 PM](https://user-images.githubusercontent.com/1937582/191937797-6068db96-86f4-4e9f-a2b6-83558145fbf1.png)

#### Before
![Screen Shot 2022-09-23 at 4 48 47 PM](https://user-images.githubusercontent.com/1937582/191937813-f0743f74-84a2-4f84-b40b-7297a6b3dc64.png)